extensions/stackdriver/stackdriver.h

```diff
@@ -114,19 +120,52 @@ class StackdriverRootContext : public RootContext {
       kDefaultEdgeEpochReportDurationNanoseconds;
 
   bool use_host_header_fallback_;
+
+  std::unordered_map<uint32_t, std::shared_ptr<::Wasm::Common::RequestInfo>>
+      tcp_request_queue_;
 };
 
 // StackdriverContext is per stream context. It has the same lifetime as
 // the request stream itself.
 class StackdriverContext : public Context {
  public:
-  StackdriverContext(uint32_t id, RootContext* root) : Context(id, root) {}
+  StackdriverContext(uint32_t id, RootContext* root)
+      : Context(id, root), is_tcp_(false), context_id_(id) {
+    request_info_ = std::make_shared<::Wasm::Common::RequestInfo>();
+  }
   void onLog() override;
 
+  FilterStatus onNewConnection() override {
+    is_tcp_ = true;
+    request_info_->tcp_connections_opened++;
```

Comment:
I'd prefer not let stream context own request_info_. Code is clear that way instead of passing around. You can call `addToTCPRequestQueue` and let root context initialize a request info there.

Reply:
@bianpengyuan : I am open to suggestions but why is that? I feel it belongs here, because we edit, size and connection total metrics in stream context. It's lifecycle is tied to stream's lifecycle this way...

Reply:
I am hoping not to pass the pointer around if it could belong to one place. HTTP request info could be construct on demand in root context. TCP connection info is already stored as a vector in root context, and stream context could just call root context functions to increment different fields.

This code used to store request info in each stream and it was removed in favor of current way for code simplicity.

Reply:
Fixed..PTAL